src/Runner/Filter/ChunkIterator.php

```diff
@@ -0,0 +1,75 @@
+<?php declare(strict_types=1);
+/*
+ * This file is part of PHPUnit.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace PHPUnit\Runner\Filter;
+
+use PHPUnit\Framework\TestSuite;
+use PHPUnit\Framework\WarningTestCase;
+use PHPUnit\Util\RegularExpression;
+use RecursiveFilterIterator;
+use RecursiveIterator;
+
+/**
+ * @internal This class is not covered by the backward compatibility promise for PHPUnit
+ */
+final class ChunkIterator extends RecursiveFilterIterator
+{
+    /**
+     * @var int
+     */
+    private $chunk;
+
+    /**
+     * @var int
+     */
+    private $numChunks;
+
+    /**
+     * @var int
+     */
+    private $numTests;
+
+    /**
+     * @var int
+     */
+    private $numTestsPerChunk;
+
+    /**
+     * @var int
+     */
+    private $chunkStart;
+
+    /**
+     * @var int
+     */
+    private $chunkEnd;
+
+    /**
+     * @throws \Exception
+     */
+    public function __construct(RecursiveIterator $iterator, array $args)
+    {
+        parent::__construct($iterator);
+
+        $this->chunk = $args['chunk'];
+        $this->numChunks = $args['numChunks'];
+        $this->numTests = count($iterator);
+        $this->numTestsPerChunk = floor($this->numTests / $this->numChunks);
```

Comment:
Maybe we can use `array_chunk` here to simplify the logic. Let php chunk it and then in `accept()` we can check if the current test is in the chunk being ran.